### PR TITLE
glscopeclient: Check WIXPATH before creating MSI installer

### PR DESCRIPTION
**Pull request by Biswa96**

### CODE REVIEW

**Comment by azonenberg** [comment]
The CI build failed. Do you need to install additional software in the CI environment perhaps? (see the .github directory for the scripts)

**Comment by Biswa96** [comment]
Done. @umarcor

File: msys2/PKGBUILD [hunk 1]
```diff
@@ -36,17 +35,21 @@ build() {
...
+  CXXFLAGS+=" -Wno-format" \
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  ${MINGW_PREFIX}/bin/cmake.exe \
+    -G "Ninja" \
     -DCMAKE_INSTALL_PREFIX="${MINGW_PREFIX}" \
     -DBUILD_TESTING=OFF \
     ../
-  cmake --build .
+
+  ${MINGW_PREFIX}/bin/cmake.exe --build ./
```
Comment (umarcor) [inline msys2/PKGBUILD:47]: `cmake --build ./` should work, isn't it?

File: msys2/PKGBUILD [hunk 2]
```diff
@@ -36,17 +35,21 @@ build() {
...
 }
 
 package() {
   cd "${srcdir}"/../../build
-  make DESTDIR="${pkgdir}" install
+  DESTDIR="${pkgdir}" ${MINGW_PREFIX}/bin/cmake.exe --build ./ --target install
```
Comment (umarcor) [inline msys2/PKGBUILD:52]: `DESTDIR="${pkgdir}" cmake --build ./ --target install`?

File: src/glscopeclient/CMakeLists.txt [hunk 3]
```diff
@@ -171,21 +171,19 @@ if(WIN32)
...
-	else()
-		message("Skipping MSI package build; define WIXPATH to enable")
-	endif()
+	configure_file(wix/glscopeclient.wxs.in ${CMAKE_BINARY_DIR}/glscopeclient.wxs)
+	add_custom_target(
+		msi_windows_x64 ALL
+		COMMENT "Creating Windows x64 MSI..."
+		DEPENDS dist_windows_x64
+		WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+	COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/src/glscopeclient/wix/LICENSE.rtf ${CMAKE_BINARY_DIR}
```
Comment (umarcor) [inline src/glscopeclient/CMakeLists.txt:180]: It seems that indentation is not correct in this line.

**Review by umarcor — COMMENTED** [review]
Apart from the style issues (see comments), LGTM!

File: msys2/PKGBUILD (hunk 1 above)
Comment (Biswa96) [reply msys2/PKGBUILD:47]: Yes, previous command works. The new one is just to play safely.

File: msys2/PKGBUILD (hunk 2 above)
Comment (Biswa96) [reply msys2/PKGBUILD:52]: Yes, previous command works. The new one is just to play safely.

File: src/glscopeclient/CMakeLists.txt (hunk 3 above)
Comment (Biswa96) [reply src/glscopeclient/CMakeLists.txt]: Done 👍

**Review by umarcor — APPROVED** [review]
Nice work!

**Comment by Biswa96** [comment]
One litte question. Why does the CI compile the program twice?

**Comment by azonenberg** [comment]
> One litte question. Why does the CI compile the program twice?

It should build once for Linux and once for Windows. If there's multiple builds, then we've got a bug in the scripts.

**Comment by umarcor** [comment]
@azonenberg, on Windows it is built twice:

- https://github.com/azonenberg/scopehal-apps/blob/master/.github/workflows/build.yml#L146-L149 generates an MSYS2 package (a `.zst` tarball).
- https://github.com/azonenberg/scopehal-apps/blob/master/.github/workflows/build.yml#L162-L171 generates a portable zipfile with an installer.

I guess @Biswa96 is asking why don't we extract the `.zst` for generating the zipfile, instead of creating it from scratch. I guess the answer is that we want to keep the portable zipfile build procedure standalone, so contributos can ignore the PKGBUILD recipe if they want.

**Comment by Biswa96** [comment]
As you are distributing the ZIP file to users, why not skip the msys2 package build or PKGBUILD entirely?

**Comment by umarcor** [comment]
The MSYS2/PKGBUILD build is executed for Continuous Integration/Testing purposes. That provides "nightly" artifacts for contributors to test enhancements/PRs, before the package is bumped in MSYS2 repositories.

**Comment by Biswa96** [comment]
OK, all doubt cleared 😌

**Pull request merged**